### PR TITLE
feat: add support for datadir in rpm spec

### PR DESCRIPTION
**Pull request by pmengelbert**
**What this PR does / why we need it**:

Users have requested the ability to put read-only architecture-independent data files under `/usr/share`.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Related #242 (does not completely resolve)

**Special notes for your reviewer**:

### CODE REVIEW

File: test/azlinux_test.go [hunk 1]
```diff
@@ -760,6 +760,80 @@ Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/boot
...
+					},
+				},
+			},
+			Build: dalec.ArtifactBuild{},
+			Artifacts: dalec.Artifacts{
+				Binaries: map[string]dalec.ArtifactConfig{
+					"bin": {},
+				},
+				DataFiles: map[string]dalec.ArtifactConfig{
+					"data_dir":  {},
```
Comment (adamperlin) [inline test/azlinux_test.go:810]: It might be good to add a test with a subpath in the artifact config

**Review by adamperlin — APPROVED** [review]
I think this looks fine to me. One thing to consider is, it likely would not be good practice to install directly under `%{_datadir}` as in `%{_datadir}/my_file.data`, I believe there should pretty much always be at least one directory level of nesting to specify the package the data applies to. That being said, I'm not sure here is the place to validate that necessarily.

**Comment by cpuguy83** [comment]
@adamperlin Agreed I was going to comment the same thing but yeah I don't think there's really anything we can do here other than provide guidance that it should be namespaced.

Looks like you need to `go generate ./...`

**Comment by pmengelbert** [comment]
> @adamperlin Agreed I was going to comment the same thing but yeah I don't think there's really anything we can do here other than provide guidance that it should be namespaced.
> 
> Looks like you need to `go generate ./...`

@cpuguy83 @adamperlin 
We could require a subpath for this artifact type and drop the parent directory if the `DataFile` in question is a Directory. Happy to change it to that if y'all think it would be better.

File: test/azlinux_test.go (hunk 1 above)
Comment (pmengelbert) [reply test/azlinux_test.go]: done, thanks!

**Review by cpuguy83 — CHANGES_REQUESTED** [review]
This can also use a squash.

File: spec.go [hunk 2]
```diff
@@ -128,6 +128,8 @@ type Artifacts struct {
 	Binaries map[string]ArtifactConfig `yaml:"binaries,omitempty" json:"binaries,omitempty"`
 	// Manpages is the list of manpages to include in the package.
 	Manpages map[string]ArtifactConfig `yaml:"manpages,omitempty" json:"manpages,omitempty"`
+	// DataDirs is a list of read-only architecture-independent data files, to be placed in /usr/share/
+	DataDirs map[string]ArtifactConfig `yaml:"data_files,omitempty" json:"data_files,omitempty"`
```
[the remark this answers is not in the document]
Comment (pmengelbert) [reply spec.go]: done

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**